File: LeetCode/Python/Easy/1544-Make-The-String-Great/solution.py
class Solution:
    def makeGood(self, s: str) -> str:
        result=""

        stack=list()

        for character in s:
            if not stack:
                stack.append(character)
            else:
                if (stack[-1].isupper() and character.islower()) or (character.isupper() and stack[-1].islower()):
                    if stack[-1].lower()==character.lower():
                        stack.pop()
                    else:
                        stack.append(character)
                else:
                    stack.append(character)

        result="".join(stack)

        return result
        

File: LeetCode/Rust/Easy/1544-Make-The-String-Great/solution.rs
impl Solution
{
    pub fn make_good(s: String) -> String
    {
        let mut stack: Vec<char>=Vec::new();

        for character in s.chars()
        {
            if stack.is_empty()
            {
                stack.push(character);
            }
            else
            {
                let top_of_stack: char=*(stack.last().unwrap());
                
                if (top_of_stack.is_ascii_lowercase() && character.is_ascii_uppercase()) || (top_of_stack.is_ascii_uppercase() && character.is_ascii_lowercase())
                {
                    if top_of_stack.to_ascii_lowercase()==character.to_ascii_lowercase()
                    {
                        stack.pop();
                    }
                    else
                    {
                        stack.push(character);
                    }
                }
                else
                {
                    stack.push(character);
                }
            }
        }

        stack.into_iter().collect::<String>()
    }
}